kernel/syscall.c

```diff
@@ -466,16 +466,6 @@ long myst_syscall_open(const char* pathname, int flags, mode_t mode)
     if (!(locals = malloc(sizeof(struct locals))))
         ERAISE(-ENOMEM);
 
-    if (flags & O_NOFOLLOW)
```

Comment:
Is there any reason to move this check into the filesystems themselves?

Reply:
myst_syscall_open() can be used a create a file. lstat() on a file not created yet false-trigger the ELOOP error related to the O_NOFOLLOW/O_PATH check.